src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -245,21 +245,41 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
         return configs
     }
 
+    /**
+     * Necessary to get the debug configuration over to the resolveDebugConfigurationWithSubstitutedVariables function below
+     *
+     * @param folder Workspace folder
+     * @param config User-provided config (from launch.json)
+     * @param token  Cancellation token
+     */
+    public resolveDebugConfiguration(
+        folder: vscode.WorkspaceFolder | undefined,
+        config: AwsSamDebuggerConfiguration,
+        token?: vscode.CancellationToken
+    ): vscode.ProviderResult<AwsSamDebuggerConfiguration> {
+        return config
+    }
+
     /**
      * Generates a full run-config from a user-provided config, then
      * runs/debugs it (essentially `sam build` + `sam local invoke`).
      *
      * If `launch.json` is missing, attempts to generate a config dynamically.
      *
-     * @param folder  Workspace folder
+     * TECHNICALLY this isn't the correct way to implement a runner...
+     * We should be returning a vscode.ProviderResult<AwsSamDebuggerConfiguration>.
+     * That should be picked up by a vscode.DebugAdapter (https://www.npmjs.com/package/vscode-debugadapter).
+     * The adapter should contain the execution code (this.makeConfig and this.invokeConfig).
```

Comment:
This is mentioned already here: https://github.com/aws/aws-toolkit-vscode/blob/b71260d7acbaee8ab647c0e2b70caaf5f67e4d96/src/shared/sam/debugger/awsSamDebugger.ts#L288

Let's leave out the big explanation in the docstring here. We are tracking it in the backlog.